pandas/_libs/groupby.pyx

```diff
@@ -163,14 +164,21 @@ def group_cumprod_float64(float64_t[:, :] out,
                 if val == val:
                     accum[lab, j] *= val
                     out[i, j] = accum[lab, j]
+                if val != val:
+                    if skipna:
+                        out[i, j] = NaN
+                    else:
+                        accum[lab, j] = NaN
+                        out[i, j] = accum[lab, j]
```

Comment:
Can you break early here? If you ever hit a `NaN` in a group, final output will be NaN so no need to continue adding to `NaN`.

Reply:
That's true - I added a break there.